src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -155,7 +155,19 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
         /// <summary>
         /// Gets the list of EDM navigation properties to be expanded in the response.
         /// </summary>
-        public IDictionary<IEdmNavigationProperty, SelectExpandClause> ExpandedNavigationProperties { get; private set; }
+        public IDictionary<IEdmNavigationProperty, SelectExpandClause> ExpandedNavigationProperties
+        {
+            get
+            {
+                return ExpandedNavigationPropertiesWithExpandedItem.ToDictionary(item => item.Key,
+                    item => item.Value != null ? item.Value.SelectAndExpand : null);
+            }
+        }
+
+        /// <summary>
+        /// Gets the list of EDM navigation properties to be expanded in the response.
+        /// </summary>
+        internal IDictionary<IEdmNavigationProperty, ExpandedNavigationSelectItem> ExpandedNavigationPropertiesWithExpandedItem { get; private set; }
```

Comment:
>ExpandedNavigationPropertiesWithExpandedItem [](start = 83, length = 44)

Name is pretty gross. Maybe something like ExpandedNavigationInfo, ExpandedProperties, or just ExpandedNavigations?  In 8.x we should remove ExpandedNavigationProperties and just have this new property.  We should open an issue to track the cleanup we want to do in 8.x (that would be a breaking change to do in 7.x).